### ISSUE TITLE
Some of the code may be wrong.

### ISSUE THREAD

**Issue by Mike001-wq** [Sharath-girish/efficientgaussian#6]
Should lines 234 and 235 in utils/general_utils.py be swapped? First record the val with the lowest probability, then replace it with the value with the highest probability.
`       
 

            weight[weight == val] = unique_vals[counts.argmax()]

            self.tail_locs[val.item()] = torch.where(weight == val)[0].detach().cpu()`

**Comment by Sharath-girish** [comment]
Thanks for highlighting this bug. It is now fixed in a recent [commit (1cfc84d)](https://github.com/Sharath-girish/efficientgaussian/commit/1cfc84dfb0596b54e89290388824667a4e68358b). Feel free to reopen if there are any issues.
